sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CompactibleFileStreamLog.scala

```diff
@@ -106,10 +106,8 @@ abstract class CompactibleFileStreamLog[T <: AnyRef : ClassTag](
     interval
   }
 
-  /**
-   * Filter out the obsolete logs.
-   */
-  def compactLogs(logs: Seq[T]): Seq[T]
+  /** Determine whether the log should be retained or not. */
+  def shouldRetain(log: T): Boolean
```

Comment:
@HeartSaVioR . Can we have `def shouldRetain(log: T): Boolean = true` to simplify?
`true` is reasonable for the default value for `Log`, isn't it? And, in this PR, all derived class overrides with `true` only.

Reply:
We'll enable the condition on file stream source and sink via #28363 and #28422 so eventually majority of derived classes will override the method. But I also agree `true` is the reasonable default value. I'll update this. Thanks!